### ISSUE TITLE
🎭 Implement automated artist alias detection with fuzzy matching

### ISSUE THREAD

**Issue by adrianwedd** [adrianwedd/ordr.fm#76]
## Overview
Enhance the existing artist alias system (#24) with automated fuzzy matching to detect artist aliases without manual configuration.

## Current State
- Basic exact-match alias resolution implemented
- Manual ARTIST_ALIAS_GROUPS configuration required
- Limited to pre-configured artist relationships

### Fuzzy Matching Algorithm
- [ ] **String similarity matching** - Levenshtein distance, Jaro-Winkler
- [ ] **Phonetic matching** - Soundex, Metaphone for pronunciation similarity
- [ ] **Token-based matching** - Handle multi-word artist names
- [ ] **Unicode normalization** - Handle accented characters and symbols
- [ ] **Abbreviation detection** - "BT" vs "Brian Transeau"

### Machine Learning Enhancement
- [ ] **Training data collection** - Use MusicBrainz artist aliases as training set
- [ ] **Feature engineering** - String similarity, phonetic features, token analysis
- [ ] **Confidence scoring** - Probability-based alias detection
- [ ] **Model training** - Siamese neural network for alias detection
- [ ] **Evaluation metrics** - Precision, recall, F1-score measurement

### Smart Detection Logic
- [ ] **Artist frequency analysis** - Prioritize artists with multiple releases
- [ ] **Label-based grouping** - Artists on same electronic labels
- [ ] **Genre consistency** - Similar genres suggest same artist
- [ ] **Time period analysis** - Active periods suggest artist identity
- [ ] **Collaboration detection** - Frequent collaborations as alias hints

### Phase 1: Basic Fuzzy Matching
- [ ] **String similarity functions** - Implement core matching algorithms
- [ ] **Configurable thresholds** - Adjustable similarity thresholds
- [ ] **Candidate detection** - Find potential alias candidates
- [ ] **User confirmation interface** - UI for confirming detected aliases
- [ ] **Manual override system** - Allow manual confirmation/rejection

### Phase 2: ML-Powered Detection  
- [ ] **Training pipeline** - Automated model training
- [ ] **Feature extraction** - Advanced feature engineering
- [ ] **Model deployment** - Integration with main system
- [ ] **Continuous learning** - Model improvement from user feedback
- [ ] **Performance optimization** - Fast inference for large collections

### Phase 3: Production Integration
- [ ] **Batch processing** - Analyze entire collections for aliases
- [ ] **Real-time detection** - Detect aliases during import
- [ ] **Web interface** - Alias management dashboard
- [ ] **API endpoints** - Programmatic alias management
- [ ] **Export/import** - Share alias configurations

### Core Algorithm
```python
def detect_artist_aliases(artists, threshold=0.85):
    aliases = []
    for i, artist1 in enumerate(artists):
        for artist2 in artists[i+1:]:
            similarity = calculate_similarity(artist1, artist2)
            if similarity >= threshold:
                aliases.append({
                    'artist1': artist1,
                    'artist2': artist2, 
                    'similarity': similarity,
                    'confidence': calculate_confidence(artist1, artist2)
                })
    return aliases
```

### Integration Points
- [ ] **Database schema** - Store similarity scores and confidence
- [ ] **Configuration updates** - Auto-generate ARTIST_ALIAS_GROUPS
- [ ] **Processing pipeline** - Integrate with album processing
- [ ] **Visualization** - Show alias relationships in dashboard
- [ ] **Validation tools** - Tools to verify alias accuracy

### Electronic Music Focus
- [ ] **Label ecosystem knowledge** - Use label relationships for detection
- [ ] **Genre-specific patterns** - Electronic music naming conventions
- [ ] **Remix artist detection** - Identify original vs remix artists
- [ ] **Pseudonym databases** - Known electronic music aliases
- [ ] **Scene-specific validation** - Validate within electronic music context

## Success Metrics
- **Precision**: >90% of detected aliases are correct
- **Recall**: >80% of true aliases are detected  
- **User satisfaction**: <10% false positive rate
- **Performance**: <1 second per artist comparison
- **Coverage**: Handle 95% of electronic music alias patterns

## Testing Strategy
- [ ] **Ground truth dataset** - Known artist aliases for validation
- [ ] **A/B testing** - Compare with manual configuration
- [ ] **Edge case testing** - Unusual names, unicode, abbreviations
- [ ] **Performance testing** - Large collection processing
- [ ] **User acceptance testing** - Real user feedback

**Priority**: Medium - Important enhancement for v2.2.0
**Effort**: 2-3 weeks
